pkg/eestream/rs_test.go

```diff
@@ -479,14 +420,14 @@ func testRSProblematic(t *testing.T, tt testCase, i int, fn problematicReadClose
 	}
 }
 
-func readAll(readers []io.Reader) ([][]byte, error) {
+func readAll(readers []io.ReadCloser) ([][]byte, error) {
 	pieces := make([][]byte, len(readers))
 	errs := make(chan error, len(readers))
 	for i := range readers {
 		go func(i int) {
 			var err error
 			pieces[i], err = ioutil.ReadAll(readers[i])
-			errs <- err
+			errs <- utils.CombineErrors(err, readers[i].Close())
```

Comment:
`errs.Combine`

Reply:
oh, yes. forgot it is already in the library.